Classes/UT3Nightshade.uc

```diff
@@ -255,21 +267,37 @@ function Fire(optional float F)
 		ArmMine.Destroy();
 
 		// Undeploy.
-		CurrentState = VS_Undeploying;
-		PlaySound(UndeploySnd, SLOT_None);
-		PlayAnim('ArmRetract', 4.8, 0.2);
-		CurrentAnim = "ArmRetract";
+		ChangeDeployState(VS_Undeploying);
 	}
 	else
 		Super.Fire(F);
 }
 function Tick(float DeltaTime)
 {
+	// 45° = 8192 RUU
+	local Rotator ShoulderRotation;
+
 	// If we have ammo available and we have no mine selected, select one right away!
 	if (bHasAmmo(0) && SelectedMine == 0)
 		SelectNextMine();
 
 	CheckState();
+
+	if (CurrentState == VS_Deployed && PlayerController(Controller) != none) {
+		ShoulderRotation.Yaw = (Rotation - Controller.Rotation).Yaw % 65536;
+		if (ShoulderRotation.Yaw < 0) {
+			ShoulderRotation.Yaw += 65536;
+		}
+
+		if (ShoulderRotation.Yaw <= 32768 && ShoulderRotation.Yaw > 8192) {
+			ShoulderRotation.Yaw = 8192;
+		} else if (ShoulderRotation.Yaw > 32768 && ShoulderRotation.Yaw < 57344) {
+			ShoulderRotation.Yaw = 57344;

```

Comment:
It feels like the `Clamp` functions should be able to handle this with less text.

Reply:
The thing here is that the clamp value depends on which half of the unit circle the angle is on:
![image](https://user-images.githubusercontent.com/5787775/34481532-9df4b2fa-efaa-11e7-82b4-88d75762ff4f.png)
The only other way I could think of would be to add `8192` (`45º`) to the angle, this would give me a clamp range from `0` to `16384` (`0º` to `90º`), then after clamping I'd remove the added `8192` (`45º`). For clarity I left the if approach, but I'd gladly do it this way, in fact I prefer it way more and I can leave a comment of some sort to explain it.

Reply:
Yea, that would work. Though now we know that the clamping needs to happen at 16k and 48k.

Reply:
Yes, I'll be changing that too!